### ISSUE TITLE
Exclude paco

### ISSUE THREAD

**Issue by pohmelie** [h2non/riprova#5]
I see nothing wrong with using third party libraries, but in some cases it looks like [is-number](https://github.com/jonschlinkert/is-number/blob/master/index.js) npm package :laughing:. It means, that js is poor, since it need such third-party «patches». I don't want to think, that python is poor, since it is not.

* `paco.wraps` should be removed cause it's enough to wrap functions with `asyncio.coroutine` without inspecting it.
``` python
>>> def sync(): print("sync")
... 
>>> async def async_(): print("async")
... 
>>> asyncio.get_event_loop().run_until_complete(asyncio.coroutine(sync)())
sync
>>> asyncio.get_event_loop().run_until_complete(asyncio.coroutine(async_)())
async
>>>
```
* `paco.TimeoutLimit`, [async_timeout](https://pypi.python.org/pypi/async_timeout/1.0.0) used when you have couple of things inside `with` block. If you have one coroutine, which you want to limit, you should use [asyncio.wait_for](https://docs.python.org/3/library/asyncio-task.html?highlight=wait_for#asyncio.wait_for) and you don't need to split logic for running with or without timeout. If there is no timeout, then your timeout should be `None`.

I hope this two suggestions are strong enough to exclude `paco` from dependencies.

**Comment by h2non** [comment]
Thanks for the comments. 

> paco.wraps should be removed cause it's enough to wrap functions with asyncio.coroutine without inspecting it.

I agree about the coroutine wraps is not necessary. Code was simplified now.

> paco.TimeoutLimit, async_timeout used when you have couple of things inside with block.

I'm not sure about this. You can't always timeout limit coroutines via `asyncio.wait_for`, and that's why I'm using `paco`. Even running a single coroutine but that coroutine can spawn more coroutines. Since `riprova` is be designed to support a variety of domain-agnostic scenarios, I think the current approach would be more versatile, safer and faster, so I don't think I'm going to remove that for now.

I'd prefer keeping things dependency-free if possible. That makes it more lightweight, a bit fearless for conservative developers and therefore more open to embed across projects.
I can consider moving the `paco` timeout logic into `riprova` in the future, but at the end `paco` is a small dependency without dependencies itself, so I don't see this as a big deal. 

I suffered for years the obsession about overdependencies in the JS ecosystem, but I personally don't defend the radical opposite pattern either.

**Comment by pohmelie** [comment]
> You can't always timeout limit coroutines via `asyncio.wait_for`

Example

> Even running a single coroutine but that coroutine can spawn more coroutines

What mean `spawn` in this context? If my coroutine create extra task both (`paco` and `wait_for`) will not cancel them, cause they don't know nothing about them. If you talk about `await` inside coroutine, then both will cancel all of them. If you look at source code of [`paco.TimeoutLimit`](https://github.com/h2non/paco/blob/master/paco/timeout.py#L45) and [`asyncio.wait_for`](https://github.com/python/asyncio/blob/f9b0d516fa60f9da35e87da344e365f604281ccf/asyncio/tasks.py#L364) you will see they are pretty same.

**Comment by h2non** [comment]
> Example

[aiohttp](https://github.com/KeepSafe/aiohttp) does and intensive use of that timeout approach, as well as some [aio-libs](https://github.com/aio-libs).

I'm open to rely on the `stdlib` for timeout limit, but I want to make sure it's capable of handling a wide range of cases accordingly. I will investigate a bit since I don't all the answers yet.

**Comment by pohmelie** [comment]
> aiohttp does and intensive use of that timeout approach, as well as some aio-libs.

Sure, because async-timeout was born as aiohttp helper and later moved to separate package. And it used just as sugar to not create extra coroutine for `asyncio.wait_for`. So, in this case, where it is exactly one coroutine there is no reasons to use this context manager.

**Comment by h2non** [comment]
Shipped in `v0.2.0`. Closing.
